File: GARI_Umbria_2026-08-03/DotNet/AgronicaCoreStd/AgronicaCoreModelsSTD/attivita/Lavorazione.cs
using AgronicaCoreModelsSTD.attivita.categorie;
using System;
using System.Collections.Generic;
using System.Text;

namespace AgronicaCoreModelsSTD.attivita
{
    public class Lavorazione : Job
    {
        public CategoriaOperazione categoriaOperazione { get; set; }

        public Lavorazione()
        {
            tipo = TipiJob.LAVORAZIONE;
            primaryKey = new PK(costanti.ClassType.Lavorazione, "0");
        }

        public Lavorazione(int codice)
        {
            tipo = TipiJob.LAVORAZIONE;
            primaryKey = new PK(costanti.ClassType.Lavorazione, codice.ToString());
        }

        public Lavorazione(int codice, string descrizione)
        {
            tipo = TipiJob.LAVORAZIONE;
            this.primaryKey = new PK(costanti.ClassType.Lavorazione, codice.ToString());
            this.descrizione = descrizione;
        }

    }
}

File: GARI_Umbria_2026-08-03/GiasNG/src/app/Model/attivita/Lavorazione.ts
import {CategoriaOperazione} from './categorie/CategoriaOperazione';
import {Job, PKJob} from './Job';
import {TipiJob} from './TipiJob';

export class Lavorazione extends Job {
  categoriaOperazione: CategoriaOperazione;
  descrizione: string | null = null;

  constructor(codice: string) {
    super({classType: 'Lavorazione', codice: codice} as PKJob);
  }

  getTipo(): TipiJob {
    return TipiJob.LAVORAZIONE;
  }
}
